config/initializers/03_sidekiq.rb

```diff
@@ -11,9 +11,18 @@
     config.redis = redis_config
 
     config.death_handlers << ->(job, original_exception) do
-      if original_exception.is_a?(Pender::Exception::RetryLater)
-        limit_hit_exception = Pender::Exception::RetryLimitHit.new(original_exception)
-        PenderSentry.notify(limit_hit_exception, { job: job, original_exception: original_exception.cause.inspect })
+      case original_exception
+      when Pender::Exception::RetryLater
+        if original_exception.message.include?("Too Many Requests")
+          rate_limit_exception = Pender::Exception::RateLimitExceeded.new(original_exception)
+          PenderSentry.notify(rate_limit_exception, { job: job, original_exception: original_exception.cause.inspect })
+        else
+          limit_hit_exception = Pender::Exception::RetryLimitHit.new(original_exception)
+          PenderSentry.notify(limit_hit_exception, { job: job, original_exception: original_exception.cause.inspect })
+        end
+      when Pender::Exception::RetryLimitHit
```

Comment:
Do we really need this `when` here? Is it possible for this exception to be a `original_exception`?

Reply:
Now it will catch all `RetryLimit` errors, and any other errors will be handled below without the `rate_limited: true` flag.